contrib/mmap/src/generator.cpp

```diff
@@ -220,6 +232,14 @@ bool handleArgs(int argc, char** argv,
 
             offMeshInputPath = param;
         }
+        else if (strcmp(argv[i], "--workdir") == 0)
+        {
+            param = argv[++i];
```

Comment:
You're not checking if there's an argument after i. If the command ends with ```... --workdir``` that's an out of bounds access.

Reply:
oh you're right - looks like that counts to all multivalue params. I'll fix that

Reply:
@evil-at-wow meanwhile I had a look once again into that and I assume the loop is ending one before end of `argv` array. that's why other arguments hadn't that before. I would keep the current state and rework that later if necessary are you agree with that?

Reply:
I checked again, and assuming I've been looking at the correct commit, it looks like it's just skipping the first argument (which makes sense to skip when looking at options, because that's the program name):
```
for (int i = 1; i < argc; ++i)
...
```
So when ```i``` gets its maximum value (being one less than argc) then ```argv[i]``` (=```argv[argc-1]```) is the last element of argv (remember that in C/C++ array indexes start at 0) and ```argv[++i]]``` is one past that, so an out of bounds access.

So I think they all have it. Try running the program with a missing value for an option at the end (say ```... --tile``` or ```... --maxAngle```) and see what happens.